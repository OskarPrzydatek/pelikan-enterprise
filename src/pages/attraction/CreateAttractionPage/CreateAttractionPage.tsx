import { SubmitHandler, useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';

import { fetchPost } from '~/api';
import { CreateAttractionTemplate } from '~/components/templates';
import { Endpoints, Slugs } from '~/constants';
import { IAttraction } from '~/models';
import { errorNotification, successNotification } from '~/notifications';

export const CreateAttractionPage: React.FC = () => {
  const methods = useForm<IAttraction>();
  const navigate = useNavigate();

  const onSubmit: SubmitHandler<IAttraction> = async (data) => {
    const response = await fetchPost<IAttraction>(
      Endpoints.CREATE_ATTRACTION,
      data
    );

    if (response.ok) {
      successNotification('Atrakcję dodano pomyślnie');
      navigate(`/${Slugs.ATTRACTION_OVERVIEW}`);
      return;
    }

    errorNotification();
  };

  return (
    <CreateAttractionTemplate
      createAttractionFormMethods={methods}
      onSubmitCreateAttraction={methods.handleSubmit(onSubmit)}
    />
  );
};
