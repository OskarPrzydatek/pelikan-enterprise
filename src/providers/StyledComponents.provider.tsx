import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
