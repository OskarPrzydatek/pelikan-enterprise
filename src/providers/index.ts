export { StyledComponentsProvider } from './StyledComponents.provider';
