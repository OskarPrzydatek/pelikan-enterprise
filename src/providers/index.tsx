import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
