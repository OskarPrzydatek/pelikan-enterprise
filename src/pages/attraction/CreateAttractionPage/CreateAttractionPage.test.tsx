import { render } from '@testing-library/react';
import { describe } from 'vitest';

import { TestingPageProvider } from '~/providers';

import { CreateAttractionPage } from './CreateAttractionPage';

const MockCreateAttractionPage: React.FC = () => (
  <TestingPageProvider element={<CreateAttractionPage />} />
);

describe('CreateAttractionPage', () => {
  test('page snapshot', () => {
    const view = render(<MockCreateAttractionPage />);
    expect(view).toMatchSnapshot();
  });
});
