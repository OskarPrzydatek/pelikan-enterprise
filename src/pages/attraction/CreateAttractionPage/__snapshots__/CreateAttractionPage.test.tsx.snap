// Vitest Snapshot v1

exports[`CreateAttractionPage > page snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR cJrdyV"
      >
        <form
          class="sc-jIRcFI iwFYYu"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Dodaj atrakcję
          </h2>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <input
                class="sc-fLcnxK gGYQKp"
                data-testid="create-attraction-name"
                id="name"
                name="name"
                placeholder="Nazwa atrakcji"
              />
            </div>
          </div>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <input
                class="sc-fLcnxK gGYQKp"
                data-testid="create-attraction-price"
                id="price"
                name="price"
                placeholder="Cena"
              />
            </div>
          </div>
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <textarea
                class="sc-hHTYSt lavPOm"
                data-testid="attraction-description"
                id="description"
                name="description"
                placeholder="Opis"
              />
            </div>
          </div>
          <input
            class="sc-hhOBVt dvAfkB"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR cJrdyV"
    >
      <form
        class="sc-jIRcFI iwFYYu"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Dodaj atrakcję
        </h2>
        <div
          class="sc-dmctIk iThEeJ"
        >
          <div
            class="sc-kgTSHT fmvgAY"
          >
            <input
              class="sc-fLcnxK gGYQKp"
              data-testid="create-attraction-name"
              id="name"
              name="name"
              placeholder="Nazwa atrakcji"
            />
          </div>
        </div>
        <div
          class="sc-dmctIk iThEeJ"
        >
          <div
            class="sc-kgTSHT fmvgAY"
          >
            <input
              class="sc-fLcnxK gGYQKp"
              data-testid="create-attraction-price"
              id="price"
              name="price"
              placeholder="Cena"
            />
          </div>
        </div>
        <div
          class="sc-bjfHbI kTUcsA"
        >
          <div
            class="sc-idXgbr enkruV"
          >
            <textarea
              class="sc-hHTYSt lavPOm"
              data-testid="attraction-description"
              id="description"
              name="description"
              placeholder="Opis"
            />
          </div>
        </div>
        <input
          class="sc-hhOBVt dvAfkB"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
